lightning/src/ln/channelmanager.rs

```diff
@@ -3032,6 +3038,8 @@ impl<M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelManager<M, T, K, F
 			mem::swap(&mut forward_htlcs, &mut self.forward_htlcs.lock().unwrap());
 
 			for (short_chan_id, mut pending_forwards) in forward_htlcs {
+				let mut channel_state_lock = self.channel_state.lock().unwrap();
```

Comment:
Had to revert the movement of `channel_state` into the `if` which was introduced with #1772 / https://github.com/lightningdevkit/rust-lightning/commit/f0c6dfbd80ea5a1166f711c919318a28e87e3c92. Let me know if that's fine.

Reply:
As-is this also may result in some changes to the lock order dependency graph (in top-level `ChannelManager` docs), though we want to get rid of those docs at some point anyway

Reply:
Alright, now reverted the reverted move and added `prev_user_channel_id` to `PendingAddHTLCInfo` with https://github.com/lightningdevkit/rust-lightning/pull/1856/commits/f123105f49624d6c111bb5271e1ecaf5b8835e10.